gpii/node_modules/ontologyHandler/src/ontologyHandler.js

```diff
@@ -106,6 +107,111 @@ URLs - this aspect is taken care of in the preferences server component.
             transformSpecs[filename] = JSON.parse(transformSpec);
         });
         that.options.ontologyTransformSpecs = transformSpecs;
+        // Generate Ontology Routes
+        var edges = [];
+        fluid.each(transformSpecs, function(spec, key) {
+            // The transforms are bidirectional so we add both directed edges
+            edges.push(key.split('-'), key.split('-').reverse());
+        });
+        that.options.ontologyTransformRoutes = 
+            gpii.ontologyHandler.floydWarshallAlgorithmWithReconstruction(edges);
+    };
+
+    /*
+     * Implementation of the Floyd-Warshall algorithm to determine the shortest
+     * paths between a set of named edges. This is mostly generic, but currently
+     * used in generating a path between multiple linked ontologies.
+     * https://en.wikipedia.org/wiki/Floyd–Warshall_algorithm#Path_reconstruction
+     * Currently assumes that all edges are weight 1.
+     *
+     * @edges (Array) - An array of arrays of length 2, consisting of the named
+     *     edges such as [['A','B'], ['B','C']]
+     * @returns (Object) - An object containing the necessary structures to resolve
+     *     paths using the floydWarshallPath function
+     */
+    gpii.ontologyHandler.floydWarshallAlgorithmWithReconstruction = function ( edges ) {
```

Comment:
split up into code to initialize the vertice mapping, core matrices, and algorithm itself.